packages/core/src/provider/types.ts

```diff
@@ -262,9 +264,9 @@ export interface ISmartAccountProvider<
    *
    * @param fn - a function that given public rpc client, returns a smart contract account
    */
-  connect(
-    fn: (provider: PublicErc4337Client<TTransport>) => BaseSmartContractAccount
-  ): this & { account: BaseSmartContractAccount };
+  connect<TAccount extends ISmartContractAccount>(
+    fn: (provider: PublicErc4337Client<TTransport>) => TAccount
+  ): ISmartAccountProvider<TTransport>;
```

Comment:
I think this should also be
```suggestion
  ): this & { account: TAccount };
```